src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prevNode;
+        private Node<T> nextNode;
+
+        Node(Node<T> prevNode, T value, Node<T> nextNode) {
+            this.prevNode = prevNode;
+            this.value = value;
+            this.nextNode = nextNode;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        if (index == 0) {
+            final Node<T> newNode = new Node<>(null, value, head);
+            head.prevNode = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        checkIndex(index);
+        insertBefore(value, findNode(index));
```

Comment:
```suggestion
        checkIndex(index);
        Node<T> current = findNode(index);
        Node<T> newNode = new Node<>(current.prev, value, current);
        ....
```
далі потрібно просто правильно перевизначити зв'язки і так можна позбутись тут if конструкції